src/main/java/com/healthier/admin/domain/reward/service/RewardService.java

```diff
@@ -0,0 +1,95 @@
+import com.healthier.admin.domain.reward.repository.UserRewardRepository;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class RewardService {
+
+    private final RewardRepository rewardRepository;
+    private final UserRewardRepository userRewardRepository;
+
+    // 리워드 목록 조회
+    public PageResponse<?> getAllRewards(PageCondition pageCondition) {
+        Pageable pageable = PageRequest.of(pageCondition.getPage(), pageCondition.getSize());
+        Page<Reward> rewards = rewardRepository.findAll(pageable);
+        List<RewardDto> rewardResponses = rewards.stream().map(RewardDto::fromPreview).toList();
+        return new PageResponse<>(rewardResponses, rewards.getTotalElements());
+    }
+
+    // 리워드 개별 조회
+    public RewardDto getReward(Long id) {
+        return rewardRepository
+                .findById(id)
+                .map(RewardDto::fromDetail)
+                .orElseThrow(() -> new IllegalArgumentException("해당 리워드가 존재하지 않습니다."));
+    }
+
+    // 리워드 생성
+    @Transactional
+    public void createReward(RewardDto rewardDto) {
+        Reward reward = Reward.from(rewardDto);
+        rewardRepository.save(reward);
+    }
+
+    // 리워드 수정
+    @Transactional
+    public void updateReward(Long id, RewardDto rewardDto) {
+        Reward reward =
+                rewardRepository
+                        .findById(id)
+                        .orElseThrow(() -> new IllegalArgumentException("해당 리워드가 존재하지 않습니다."));
+        updateProperty(rewardDto.getType(), reward::updateType);
+        updateProperty(rewardDto.getPoint(), reward::updatePoint);
+        updateProperty(rewardDto.getActualPrice(), reward::updateActualPrice);
+        updateProperty(rewardDto.getName(), reward::updateName);
+        updateProperty(rewardDto.getDuration(), reward::updateDuration);
+        updateProperty(rewardDto.getProductNotice(), reward::updateProductNotice);
+        updateProperty(rewardDto.getRewardNotice(), reward::updateRewardNotice);
+        updateProperty(rewardDto.getImageUrl(), reward::updateImageUrl);
+        updateProperty(rewardDto.isPublic(), reward::updateIsPublic);
+
+        rewardRepository.save(reward);
+    }
+
+    // 리워드 지급 전체 조회
+    public PageResponse<?> getAllUserRewards(PageCondition pageCondition) {
+        Pageable pageable = PageRequest.of(pageCondition.getPage(), pageCondition.getSize());
+        Page<UserReward> userRewards = userRewardRepository.findIsSelected(pageable);
+        List<UserRewardResponse> userRewardResponses =
+                userRewards.stream().map(UserRewardResponse::from).toList();
```

Comment:
```suggestion
                userRewards
        .map(UserRewardResponse::from)
        .getContent();
```

**Page** 인터페이스에서 **map** 을 바로 사용할 수 있다고 합니다. 
stream 으로 변환하고 리스트로 다시 바꾸지 않아도 될 것 같아요!

Reply:
수정했는데 확인 부탁드립니당~